data_structures/cache.py

```diff
@@ -0,0 +1,74 @@
+class DLNode:
+    def __init__(self, key, data):
+        self.prev = None
+        self.next = None
+        self.key = key
+        self.data = data
+
+
+class DLList:
```

Comment:
i have no idea how to do it properly without an additional class... what the bottlenecks are with it?

Reply:
no worries, we can just put those utility methods in there, and that's totally ok.